sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -378,6 +380,47 @@ private[hive] class HiveClientImpl(
 
       val properties = Option(h.getParameters).map(_.asScala.toMap).orNull
 
+      val excludedTableProperties = Set(
+        // The following are hive-generated statistics fields. Currently, only total_size and
+        // row_count are used to populate the dedicated field `stats`.
+        // TODO: stats should include all the other three fields.
+        StatsSetupConst.COLUMN_STATS_ACCURATE,
+        StatsSetupConst.NUM_FILES,
+        StatsSetupConst.NUM_PARTITIONS,
+        StatsSetupConst.ROW_COUNT,
+        StatsSetupConst.RAW_DATA_SIZE,
```

Comment:
ah I see. So our targets are:

1. recognize hive statistics, i.e. we should set the `CatalogTable.stats` according to hive stats properties
2. don't overwrite hive stats properties.
3. SHOW CREATE TABLE shouldn't print hive stats properties.

My proposal: In `HiveClientImpl`, set `CatalogTable.stats` by hive stats properties, and still keep them in table properties. In SHOW CREATE TABLE, hide the hive stats properties.